services/vision/vision_test.go

```diff
@@ -1,52 +1,57 @@
-					Type: "classifications",
-					Parameters: utils.AttributeMap{
-						"model_path":  modelPath,
-						"label_path":  labelPath,
-						"num_threads": 1,
-					},
-				},
-			},
-		}
-	}
-
-	visionConf := makeVisionConfig("/some/path/on/robot/model.tflite", "/other/path/on/robot/textFile.txt")
-	visionConfWithRefs := makeVisionConfig("${packages.test_model}/model.tflite", "${packages.test_model}/textFile.txt")
-	visionConfOneRef := makeVisionConfig("/some/path/on/robot/model.tflite", "${packages.test_model}/textFile.txt")
+const (
+	testVisionServiceName  = "vision1"
+	testVisionServiceName2 = "vision2"
+)
 
-	packageManager := packages.NewNoopManager()
+func TestFromRobot(t *testing.T) {
+	svc1 := &inject.VisionService{}
+	svc1.DetectionsFunc = func(ctx context.Context, img image.Image, extra map[string]interface{}) ([]objectdetection.Detection, error) {
+		det1 := objectdetection.NewDetection(image.Rectangle{}, 0.5, "yes")
+		return []objectdetection.Detection{det1}, nil
+	}
+	r := &inject.Robot{}
+	r.ResourceByNameFunc = func(name resource.Name) (resource.Resource, error) {
+		return svc1, nil
+	}
+	svc, err := vision.FromRobot(r, testVisionServiceName)
+	test.That(t, err, test.ShouldBeNil)
+	test.That(t, svc, test.ShouldNotBeNil)
+	result, err := svc.Detections(context.Background(), nil, nil)
+	test.That(t, err, test.ShouldBeNil)
+	test.That(t, len(result), test.ShouldEqual, 1)
+	test.That(t, result[0].Score(), test.ShouldEqual, 0.5)
+}
 
-	testConfigWalker := func(t *testing.T, conf *Config, expectedModelPath, expectedLabelPath string) {
-		newConf, err := conf.Walk(packages.NewPackagePathVisitor(packageManager))
-		test.That(t, err, test.ShouldBeNil)
+type simpleDetector struct{}
 
-		test.That(t, newConf.ModelRegistry, test.ShouldNotBeNil)
-		test.That(t, newConf.ModelRegistry, test.ShouldHaveLength, 1)
-		test.That(t, newConf.ModelRegistry[0].Name, test.ShouldEqual, "my_classifier")
-		test.That(t, newConf.ModelRegistry[0].Type, test.ShouldEqual, "classifications")
-		test.That(t, newConf.ModelRegistry[0].Parameters, test.ShouldNotBeNil)
-		test.That(t, newConf.ModelRegistry[0].Parameters["model_path"], test.ShouldEqual, expectedModelPath)
-		test.That(t, newConf.ModelRegistry[0].Parameters["label_path"], test.ShouldEqual, expectedLabelPath)
-		test.That(t, newConf.ModelRegistry[0].Parameters["num_threads"], test.ShouldEqual, 1)
-	}
+func (s *simpleDetector) Detect(context.Context, image.Image) ([]objectdetection.Detection, error) {
+	det1 := objectdetection.NewDetection(image.Rectangle{}, 0.5, "yes")
+	return []objectdetection.Detection{det1}, nil
+}
 
-	testConfigWalker(t, visionConf, "/some/path/on/robot/model.tflite", "/other/path/on/robot/textFile.txt")
-	testConfigWalker(t, visionConfWithRefs, "test_model/model.tflite", "test_model/textFile.txt")
-	testConfigWalker(t, visionConfOneRef, "/some/path/on/robot/model.tflite", "test_model/textFile.txt")
+func TestNewService(t *testing.T) {
+	r := &inject.Robot{}
```

Comment:
better to var decl these and do a `&` later

Reply:
comment applies to other instances in this file

Reply:
Done.